### PR TITLE
CLD-848 Update role for cloud-ops

### PR DESCRIPTION
**Pull request by slam1987**
**Summary**

Revised this role to be more up-to-date with our current Ansible scheme.

Role did not work out of the box and some changes were made to work.

**Tests**

Confirmed working when HAproxy is configured correctly using PR: [https://github.com/webnifico/cloud-ops/pull/726](https://github.com/webnifico/cloud-ops/pull/726)

Confirmed that metrics are showing up:

```
stanleylam@keycloak-controller001-dev:~$ curl -s http://localhost:9101/metrics | grep "^haproxy"
haproxy_backend_bytes_in_total{backend="keycloak_be"} 0
haproxy_backend_bytes_out_total{backend="keycloak_be"} 0
haproxy_backend_connection_errors_total{backend="keycloak_be"} 0
haproxy_backend_current_queue{backend="keycloak_be"} 0
haproxy_backend_current_server{backend="keycloak_be"} 0
haproxy_backend_current_session_rate{backend="keycloak_be"} 0
haproxy_backend_current_sessions{backend="keycloak_be"} 0
haproxy_backend_http_connect_time_average_seconds{backend="keycloak_be"} 0
haproxy_backend_http_queue_time_average_seconds{backend="keycloak_be"} 0
haproxy_backend_http_response_time_average_seconds{backend="keycloak_be"} 0
haproxy_backend_http_responses_total{backend="keycloak_be",code="1xx"} 0
haproxy_backend_http_responses_total{backend="keycloak_be",code="2xx"} 0
haproxy_backend_http_responses_total{backend="keycloak_be",code="3xx"} 0
haproxy_backend_http_responses_total{backend="keycloak_be",code="4xx"} 0
haproxy_backend_http_responses_total{backend="keycloak_be",code="5xx"} 0
haproxy_backend_http_responses_total{backend="keycloak_be",code="other"} 0
haproxy_backend_http_total_time_average_seconds{backend="keycloak_be"} 0
haproxy_backend_limit_sessions{backend="keycloak_be"} 200
haproxy_backend_max_queue{backend="keycloak_be"} 0
haproxy_backend_max_session_rate{backend="keycloak_be"} 0
haproxy_backend_max_sessions{backend="keycloak_be"} 0
haproxy_backend_redispatch_warnings_total{backend="keycloak_be"} 0
haproxy_backend_response_errors_total{backend="keycloak_be"} 0
haproxy_backend_retry_warnings_total{backend="keycloak_be"} 0
haproxy_backend_sessions_total{backend="keycloak_be"} 0
haproxy_backend_up{backend="keycloak_be"} 0
...
```

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-haproxy_exporter_bin_dir: "/usr/local/bin"
+haproxy_exporter_bin_dir: /opt/haproxy-exporter
```
Comment (nikosmeds) [inline defaults/main.yml]: I know this is WIP, but any reason we're deviating from where most exporters are installed?

```
$ which node_exporter
/usr/local/bin/node_exporter
$ which mysqld_exporter
/usr/local/bin/mysqld_exporter
```

File: tasks/main.yml [hunk 2]
```diff
@@ -78,23 +75,23 @@
...
     state: "{{ haproxy_exporter_service_state }}"
+    daemon_reload: true
 
 - name: curl haproxy-exporter metrics
   uri:
     url: http://localhost:9101/metrics
     return_content: yes
   register: metrics
-  check_mode: no
+  changed_when: false
```
Comment (nikosmeds) [inline tasks/main.yml]: I don't believe this is necessary, e.g. our Keycloak tests don't include and don't show changes on each run:

```
- name: test that keycloak login page loads
  uri:
    url: "https://{{ inventory_hostname }}:8443/auth/"
    return_content: yes
    timeout: 60
  register: login_page
  failed_when: "'Welcome to Keycloak' not in login_page.content"
```

(also looks like we can combine the following task into this one)

File: defaults/main.yml (hunk 1 above)
Comment (slam1987) [reply defaults/main.yml]: No reason, changed it to match with others.

File: tasks/main.yml (hunk 2 above)
Comment (slam1987) [reply tasks/main.yml]: Removed

**Review by nikosmeds — APPROVED** [review]

**Pull request merged**